user_profile/forms.py

```diff
@@ -0,0 +1,34 @@
+from django import forms
+from django.core.validators import RegexValidator
+from .models import User, UserProfile
+from django.contrib.auth.hashers import make_password
+from .mixins import EmailDuplicationMixin
+
+
+class SignUpForm(EmailDuplicationMixin, forms.ModelForm):
```

Comment:
We probably should have two forms one for user and other for user_profile.  Using single forms complicates things when you are going to update user's profile.

Reply:
Should I change signup into separate forms or handle update user's profiles with to different forms?

Reply:
I think for both signup and update profile use cases.

Reply:
Separate forms are created for User and UserProfile. SigUp and UpdateProfile both cases are handled.